src/cpp/library/type_traits

```diff
@@ -0,0 +1,36 @@
+#ifndef STL_TYPE_TRAITS
+#define STL_TYPE_TRAITS
```

Comment:
1. Do we need this new header? What happens if we use the one from the C++ library installed on the system?
2. Please consider using `#pragma once` (our Clang frontend supports this) or to rename the include guard to have an `__ESBMC_` prefix.

Reply:
> What happens if we use the one from the C++ library installed on the system?

I tried it and our frontend complained about it. It's not just this method in the standard library, there are a lot of other methods that our frontend doesn't support yet. I thought maybe we could handle it this way for now to avoid more changes.

> Please consider using #pragma once (our Clang frontend supports this) or to rename the include guard to have an __ESBMC_ prefix.

Yes, I'll change it.